api/messages.proto

```diff
@@ -202,7 +202,10 @@ message ApplicationUplink {
 
   bool confirmed = 11;
 
-  // next: 13
+  // End device location metadata, set by the Application Server while handling the message.
+  map<string,Location> locations = 14;
```

Comment:
This is to account for https://github.com/TheThingsNetwork/lorawan-stack/pull/3298/files#diff-5e41eaf89519c9fe313dc525c2d3712dR207

Reply:
This link doesn't seem to go to a related target, can you elaborate?

Reply:
I guess he meant the `google.protobuf.Duration consumed_airtime = 13` added in #3298.